cbsiamlg/whisper-asr-webservice
# feature/AMLG-6381-update-faster-whisper-and-GPU-clean-up

### Overview
Update to faster-whisper and clean up of `pyproject` file for `pytorch`. Also, streamlined the GPU Dockerfile to not pull a wheel if not needed but use pip. 

### QA

1. `docker compose -f docker-compose.yml build`
2. ` ASR_MODEL=medium.en ASR_ENGINE=faster_whisper docker composes up`
3.  You'll see the downloads in the console. 
4. Try an audio file - here's my request: 
 ```
curl -X 'POST' \
  'http://localhost:9000/asr?task=transcribe&language=en&encode=false&output=json&word_timestamps=true' \
  -H 'accept: application/json' \
  -H 'Content-Type: multipart/form-data' \
  -F 'audio_file=@Welcome_to_Mystery_T.wav;type=audio/wav'
```
5. Got output!
6. Ctrl+C to shutdown and `docker compose down to remove`

**Bonus:** 
1. Build gpu to test that it works. `docker compose -f docker-compose.gpu.yml build` 
WARNING!
I suggest doing a cache clean up before you do this: `docker builder prune` and `docker image prune -a -f ` for previous steps. Previous build is 7.826GB.
